dxr/query.py

```diff
@@ -326,7 +326,9 @@ def visit_text(self, text, ((some_text,), _)):
         ``visit_filtered_term`` will overrule us later.
 
         """
-        return {'name': 'text', 'arg': some_text}
+        # Case-sensitive if there's any uppercase characters in the term.
+        case_sensitive = any((c.isupper() for c in some_text))
```

Comment:
Right choice! If you had required "all lowercase", unicode would have caused a disaster.